### PR TITLE
Enables Use of input_to_graph_point() by CurvesAsSubmobjects Instances

### PR DESCRIPTION
**Pull request by JinchuLi2002**
## Overview: What does this pull request change?
Related to #3150
Fixed the issue where ```input_to_graph_point()``` was unusable with curves created by ```CurvesAsSubmobjects``` ~since the latter did not have 'points' as an attribute~ by overriding ```point_from_proportion``` in CurvesAsSubmobjects, where points are created from submobjects instead of the points attribute.

## Links to added or changed documentation pages
[CurvesAsSubmobjects](https://manimce--3190.org.readthedocs.build/en/3190/reference/manim.mobject.types.vectorized_mobject.CurvesAsSubmobjects.html)
## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by JinchuLi2002** [comment]
Any idea why those automatic tests are failing? 

This is my first PR to the community, thank you for your patience and any input is appreciated.

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -2343,6 +2343,7 @@ def __init__(self, vmobject, **kwargs):
             part = VMobject()
             part.set_points(tup)
             part.match_style(vmobject)
+            self.append_points(tup)
             self.add(part)
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:2347]: I have not investigated closely, but it seems like this change includes a certain redundancy when it comes to the points of `self`: not only are the points added to the mobject itself, but they are also added as part of `self`'s submobjects. This is not good (as the mobject would, in this case, also be drawn twice on the screen).

**Review by behackl — CHANGES_REQUESTED** [review]
I'm not quite sure what the best course of action is here. I don't think it makes sense to merge this solution given the introduced point duplication (Mobjects being painted twice is a bug).

One approach that might make sense instead: the implementation of `VMobject.point_from_proportion` could be changed such that *only in the case where `self.points` is empty*, the point is constructed from the points of the submobjects instead of just raising an exception.

Or, perhaps even better: `VGroup` (or, more conservatively, just `CurvesAsSubmobjects`) could implement their own `point_from_proportion` method where the point is created from the submobjects, instead of from the interpolation of the `points` attribute.

What do you think?

File: manim/mobject/types/vectorized_mobject.py (hunk 1 above)
Comment (JinchuLi2002) [reply manim/mobject/types/vectorized_mobject.py]: Hey, thank you so much for the suggestions! I now see that implement ```CurvesAsSubmobjects```'s own method makes much more sense. I have updated the PR.

Thanks for your patience!

File: manim/mobject/types/vectorized_mobject.py [hunk 2]
```diff
@@ -2345,6 +2345,69 @@
             part.match_style(vmobject)
             self.add(part)
 
+    def point_from_proportion(self, alpha: float) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (JinchuLi2002) [reply manim/mobject/types/vectorized_mobject.py:2348]: This is consistent with VMobject.point_from_proportion()

**Review by behackl — APPROVED** [review]
Very nice! Thank you very much for this new and improved implementation!

**Pull request merged**